// vidstack/dev/chunks/vidstack-CWk48RFI.js
import { H as Host, g as effect, s as setAttribute } from './vidstack-CPyWhQNM.js';
import { P as Poster } from './vidstack-C0WpKrrj.js';
import './vidstack-CCzOi1rX.js';
import './vidstack-Ckf8G9lg.js';
import './vidstack-B7p3DDPV.js';
import './vidstack-BoQ9v9ck.js';
import './vidstack-CnaYRoc3.js';
import './vidstack-Bnoxq110.js';
import './vidstack-CwOXU-tS.js';
import './vidstack-B18nf2ni.js';
import './vidstack-1gmLGa6x.js';
import './vidstack-DM_McBs5.js';
import './vidstack-ksPACRiU.js';
import './vidstack-CSaHpIQV.js';

class MediaPosterElement extends Host(HTMLElement, Poster) {
  constructor() {
    super(...arguments);
    this._img = document.createElement("img");
  }
  static {
    this.tagName = "media-poster";
  }
  static {
    this.attrs = {
      crossOrigin: "crossorigin"
    };
  }
  onSetup() {
    this.$state.img.set(this._img);
  }
  onConnect() {
    const { src, alt, crossOrigin } = this.$state;
    if (this._img.parentNode !== this) {
      this.prepend(this._img);
    }
    effect(() => {
      setAttribute(this._img, "alt", alt());
      setAttribute(this._img, "crossorigin", crossOrigin());
      setAttribute(this._img, "src", src() || "");
    });
    effect(() => {
      const { loading, hidden } = this.$state;
      this._img.style.display = loading() || hidden() ? "none" : "";
    });
  }
}

export { MediaPosterElement };
