// vidstack/prod/chunks/vidstack-Cg_pnaNV.js
import { H as Host, g as effect, s as setAttribute } from './vidstack-Uw94Yvrg.js';
import { P as Poster } from './vidstack-CpPO9Fgt.js';
import './vidstack-DSCDp3Na.js';
import './vidstack-Bg81bcBp.js';
import './vidstack-Dchp6m-3.js';
import './vidstack-6TFk98Ec.js';
import './vidstack-VrKElWm_.js';
import './vidstack-CYUQiNtU.js';
import './vidstack-DTYXegDE.js';
import './vidstack-Du-_dEIb.js';
import './vidstack-BSXZsAhp.js';
import './vidstack-BUqeBbTQ.js';
import './vidstack-ksPACRiU.js';
import './vidstack-CSaHpIQV.js';

class MediaPosterElement extends Host(HTMLElement, Poster) {
  constructor() {
    super(...arguments);
    this.Wm = document.createElement("img");
  }
  static {
    this.tagName = "media-poster";
  }
  static {
    this.attrs = {
      crossOrigin: "crossorigin"
    };
  }
  onSetup() {
    this.$state.img.set(this.Wm);
  }
  onConnect() {
    const { src, alt, crossOrigin } = this.$state;
    if (this.Wm.parentNode !== this) {
      this.prepend(this.Wm);
    }
    effect(() => {
      setAttribute(this.Wm, "alt", alt());
      setAttribute(this.Wm, "crossorigin", crossOrigin());
      setAttribute(this.Wm, "src", src() || "");
    });
    effect(() => {
      const { loading, hidden } = this.$state;
      this.Wm.style.display = loading() || hidden() ? "none" : "";
    });
  }
}

export { MediaPosterElement };
